Newspaper page for /theguardian

jennysivapalan opened the pull request:
As part of moving newspaper pages over to dotcom I've created `/theguardian` endpoint. Following the [R2 version] (http://www.theguardian.com/theguardian) this page

* Queries the Content API to fetch results for today for the newspaper book path `theguardian/mainsection` using the query parameter `use-date=newspaper-edition` (which ensures just content that was in the paper will be returned).
* Results which have a page number of 1 will be in a `Front Page` container
* Other results are grouped by `newspaper-book-section` and ordered by page number
* Each grouped content by newspaper-book-section is transformed into a container.
* Containers are limited to 12 results, if the user wants to see more content for a container they can click through to the newspaper-book-section e.g.`theguardian/mainsection/uknews`

![screen shot 2015-11-10 at 17 23 34](https://cloud.githubusercontent.com/assets/944375/11069785/daaa3bf6-87cf-11e5-9868-851d699b73a7.png)

![screen shot 2015-11-10 at 17 23 44](https://cloud.githubusercontent.com/assets/944375/11069786/daad78fc-87cf-11e5-9cc7-fe6ade967262.png)

JustinPinner commented:
Quite a few hoops to jump through with this one then! Looking good though. Pity we have to fall back on those _._1 and _._2 references but that's expected I guess. And those Facia types... :sweat_drops:

Does it still need the standalone routes BTW?

jennysivapalan commented:
I realised after raising this PR that this isn't going to work for Sunday as it will return zero results. So another hoop to jump through. I'll look at alternatives for `_._1` and `_._2`.

johnduffell commented on applications/app/services/TodaysNewspaperQuery.scala:
```diff
@@ -0,0 +1,105 @@
+    //order booksections by first content item in each booksection
+    val pageNumberToFaciaContainer: List[(Int, BookSectionContent)] = orderedContentForBookSection.flatMap { bookSection =>
+      val pageNumberOpt = bookSection.content.headOption.flatMap(content => getNewspaperPageNumber(content))
+      pageNumberOpt.map(_ -> bookSection)
```
if you want to get rid of the _1 and _2 consider instead of using a tuple here, make a case class with two named fields and use it here

jennysivapalan replied:
I had similar approach for grouping the tag and content (which is a stepping stone to creating the BookSectionContent). I felt I was creating case classes for just one line. However if it it makes it more readable then it's I guess that is better. Will give it a go here.

jennysivapalan commented:
Added fallback for Sunday to use the Saturday edition. Also added the date to the Front Page container.

![screen shot 2015-11-11 at 11 35 28](https://cloud.githubusercontent.com/assets/944375/11090310/cc440bc0-8868-11e5-8e7d-b5336a9f8997.png)

johnduffell commented:
anyway as discussed looks good to me! :+1: feel free to add the case classes though

jennysivapalan commented:
Thanks, yep the case classes make it much more readable so have added those in too.

johnduffell commented:
agreed, looks great :+1:
